### PR TITLE
Add another example to string-comparison-net-5-plus.md

### LINKED ISSUE #28248: Example does not explain changes in IndexOf method as of .NET 6
Please add  on[ this page](https://docs.microsoft.com/en-us/dotnet/standard/base-types/string-comparison-net-5-plus) the expected output for this code snippet also for .NET 6

    string s = "Hello\r\nworld!";
    int idx = s.IndexOf("\n");
    Console.WriteLine(idx);

Because the behaviour changed for .NET 6 when using ICU: https://github.com/dotnet/runtime/pull/57078

related issue: https://github.com/dotnet/docs/issues/28218

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ff20a875-d296-266f-8e5b-a3ab32db2491
* Version Independent ID: 1788f46c-cdad-9ea5-7bd7-62c959255fd5
* Content: [Behavior changes when comparing strings on .NET 5+](https://docs.microsoft.com/en-us/dotnet/standard/base-types/string-comparison-net-5-plus)
* Content Source: [docs/standard/base-types/string-comparison-net-5-plus.md](https://github.com/dotnet/docs/blob/main/docs/standard/base-types/string-comparison-net-5-plus.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Add example from the other article
- Link to the other article.

Fixes #28248

@gewarren 
@jonadv

### CODE REVIEW

**Comment by gewarren** [comment]
@adegeo I actually specifically removed that example in https://github.com/dotnet/docs/pull/28263/files#diff-09fa17af618d49b63bdc6845711fe5df182089327fd935ce7b18a5ab8847a9a7, for the reason that there is no effective behavior change between .NET Core and .NET 6+ (due to the fact that they reverted to the previous behavior). And since the article is mainly about behavior changes due to ICU, I didn't think it was a good example to use.

**Comment by adegeo** [comment]
@gewarren According to the information I had found, there is a difference between .NET Core vs .NET 6 based on the OS version you're using if you check the code snippet result comments. I believe .NET 6 normalized it, but .NET Core behavior would match .NET 6 in one version of the OS and not match in a different.

**Comment by gewarren** [comment]
> @gewarren According to the information I had found, there is a difference between .NET Core vs .NET 6 based on the OS version you're using if you check the code snippet result comments. I believe .NET 6 normalized it, but .NET Core behavior would match .NET 6 in one version of the OS and not match in a different.

Are you saying that on non-Windows OSs, the snippet returns `-1` in .NET 5 and .NET Core, and `6` in .NET 6?

**Comment by jonadv** [comment]
@tarekgh you probably know this

File: docs/standard/base-types/string-comparison-net-5-plus.md [hunk 1]
```diff
@@ -29,8 +29,20 @@ Console.WriteLine($"{greeting.IndexOf("\0")}");
...
 // '0' when running on .NET Core 2.x - 3.x or .NET 5 (non-Windows)
 // '3' when running on .NET Core 2.x or .NET 5+ (in invariant mode)
+
+string s = "Hello\r\nworld!";
+int idx = s.IndexOf("\n");
+Console.WriteLine(idx);
+
+// The snippet prints:
+//
+// '6' when running on .NET Core 3.1 (A version before Windows 10 May 2019 Update)
```
Comment (gewarren) [inline docs/standard/base-types/string-comparison-net-5-plus.md:39]:
```suggestion
// '6' when running on .NET Core 3.1 (Windows version before Windows 10 May 2019 Update)
```

File: docs/standard/base-types/string-comparison-net-5-plus.md [hunk 2]
```diff
@@ -29,8 +29,20 @@ Console.WriteLine($"{greeting.IndexOf("\0")}");
...
+// '-1' when running on .NET 5 (Windows 10 May 2019 Update or later)
+// '6' when running on .NET 6+
```
Comment (gewarren) [inline docs/standard/base-types/string-comparison-net-5-plus.md:41]:
```suggestion
// '6' when running on .NET 6+ (all Windows and non-Windows OSs)
```

File: docs/standard/base-types/string-comparison-net-5-plus.md [hunk 3]
```diff
@@ -29,8 +29,20 @@ Console.WriteLine($"{greeting.IndexOf("\0")}");
...
 // '3' when running on .NET Core 2.x or .NET 5+ (in invariant mode)
+
+string s = "Hello\r\nworld!";
+int idx = s.IndexOf("\n");
+Console.WriteLine(idx);
+
+// The snippet prints:
+//
+// '6' when running on .NET Core 3.1 (A version before Windows 10 May 2019 Update)
+// '-1' when running on .NET 5 (Windows 10 May 2019 Update or later)
```
Comment (gewarren) [inline docs/standard/base-types/string-comparison-net-5-plus.md:40]:
```suggestion
// '-1' when running on .NET 5 or .NET Core 3.1 (non-Windows OS)
// '-1' when running on .NET 5 (Windows 10 May 2019 Update or later)
```

**Review by gewarren — APPROVED** [review]

**Comment by adegeo** [comment]
@tarekgh Hey there! Can you just review the code comment updates in this PR and make sure they're correct?

File: docs/standard/base-types/string-comparison-net-5-plus.md [hunk 4]
```diff
@@ -29,8 +29,21 @@ Console.WriteLine($"{greeting.IndexOf("\0")}");
...
 // '0' when running on .NET Core 2.x - 3.x or .NET 5 (non-Windows)
 // '3' when running on .NET Core 2.x or .NET 5+ (in invariant mode)
+
+string s = "Hello\r\nworld!";
+int idx = s.IndexOf("\n");
+Console.WriteLine(idx);
+
+// The snippet prints:
+//
+// '6' when running on .NET Core 3.1 (Windows version before Windows 10 May 2019 Update)
```
Comment (tarekgh) [inline docs/standard/base-types/string-comparison-net-5-plus.md:39]: >Windows version before Windows 10 May 2019 Update [](http://example.com/codeflow?start=38&length=49)

please remove the part "version before Windows 10 May 2019 Update". 3.1 will always give this result on any version of Windows.

**Review by tarekgh — APPROVED** [review]
added a little extra comment. Otherwise, LGTM. Thanks for getting this cleared out.

File: docs/standard/base-types/string-comparison-net-5-plus.md (hunk 4 above)
Comment (adegeo) [inline docs/standard/base-types/string-comparison-net-5-plus.md:39]:
```suggestion
// '6' when running on .NET Core 3.1
```

**Pull request merged**